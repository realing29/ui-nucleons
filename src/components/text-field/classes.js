import classnames from 'classnames/bind';
import classes from './text-field.scss';

export const cx = classnames.bind(classes);
