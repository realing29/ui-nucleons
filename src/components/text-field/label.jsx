import React from 'react';
import { cx } from './classes';
import PropTypes from 'prop-types';

/**
 * Компонент label для TextField.
 * @param {Object} props Свойства.
 * @param {boolean} props.disabled Отключено ли поле.
 * @param {boolean} props.focused Сфокусировано ли поле.
 * @param {boolean} props.failed Ошибочно ли введено значение в поле.
 * @param {'desktop'|'mobile'} props.variant Вариант отображения поля.
 * @param {boolean} props.asPlaceholder Нужно ли отображать label как placeholder.
 * @param {string} props.children Содержимое.
 * @return {ReactElement} Компонент label.
 */
export const Label = ({
  disabled,
  focused,
  failed,
  variant,
  asPlaceholder,
  children,
}) => (
  <label
    className={cx([
      'label',
      asPlaceholder && 'as-placeholder',
      disabled && 'disabled',
      failed && 'failed',
      focused && 'focused',
      variant && `variant-${variant}`,
    ])}
    children={children}
  />
);

Label.propTypes = {
  /**
   * Отключено ли поле.
   */
  disabled: PropTypes.bool,

  /**
   * Сфокусировано ли поле.
   */
  focused: PropTypes.bool,

  /**
   * Ошибочно ли введено значение в поле.
   */
  failed: PropTypes.bool,

  /**
   * Вариант отображения поля.
   */
  variant: PropTypes.oneOf(['desktop', 'mobile']),

  /**
   * Нужно ли отображать label как placeholder.
   */
  asPlaceholder: PropTypes.bool,

  /**
   * Содержимое.
   */
  children: PropTypes.string,
};
