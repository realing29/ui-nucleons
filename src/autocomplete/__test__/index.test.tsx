import React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import { act, Simulate } from 'react-dom/test-utils';
import { Autocomplete } from '..';
import { TextField } from '../../text-field';

describe('Autocomplete', () => {
  const findMenu = (w: ReactWrapper) => w.find('div[data-testid="autocomplete:menu"]');

  const findMenuItems = (w: ReactWrapper) => w
    .find('div[data-testid="autocomplete:menu"]')
    .find('div[role="menuitem"]');

  const isCheckedItem = (w: ReactWrapper) => w.getDOMNode().classList.contains('checked');

  const openMenu = (w: ReactWrapper) => {
    act(() => {
      Simulate.click(w.find(TextField).find('[data-testid="text-field:block"]').getDOMNode());
    });
    w.update();
  };

  it('should renders correctly', () => {
    const wrapper = mount(
      <Autocomplete
        value='Hello'
        items={[
          'U.S.',
          'France',
          'China',
          'Cambodia',
          'Chile',
          'Canada',
          'Poland',
          'Russia',
          'Belarus',
          'Ukraine',
        ]}
        renderItem={item => item.toUpperCase()}
      />
    );

    expect(wrapper.getDOMNode()).toMatchSnapshot();

    // preset "filled-only-list"
    wrapper.setProps({ preset: 'filled-only-list' });
    expect(wrapper.getDOMNode()).toMatchSnapshot();

    // with menu opened
    openMenu(wrapper);
    expect(wrapper.getDOMNode()).toMatchSnapshot();

    // loading state
    wrapper.setProps({ loading: true });
    expect(wrapper.getDOMNode()).toMatchSnapshot();
  });

  it('should handle "renderItem" prop missing', () => {
    const wrapper = mount(
      <Autocomplete
        value='Hello'
        items={[
          'aaaa',
          'bbbb',
          'cccc',
          'dddd',
          'eeee',
        ]}
      />
    );

    openMenu(wrapper);
    expect(wrapper.getDOMNode()).toMatchSnapshot();
  });

  it('should handle field change event', () => {
    const spy = jest.fn();

    const wrapper = mount(
      <Autocomplete
        value='ffff'
        items={[
          'aaaa',
          'bbbb',
          'cccc',
          'dddd',
          'eeee',
        ]}
        onChange={spy}
      />
    );

    expect(findMenu(wrapper)).toHaveLength(0);
    expect(spy).toBeCalledTimes(0);

    act(() => {
      Simulate.change(
        wrapper.find(TextField).find('input[data-testid="text-field:field"]').getDOMNode(),
        { target: { value: 'jjjj' } as any }
      );
    });
    wrapper.update();

    expect(findMenu(wrapper)).toHaveLength(1);
    expect(spy).toBeCalledTimes(1);
  });

  it('should handle defaultValue', () => {
    const wrapper = mount(
      <Autocomplete
        defaultValue='dddd'
        items={[
          'aaaa',
          'bbbb',
          'cccc',
        ]}
      />
    );

    expect(wrapper.getDOMNode()).toMatchSnapshot();
  });

  it('should handle outside click', () => {
    const wrapper = mount(
      <Autocomplete
        defaultValue='dddd'
        items={[
          'aaaa',
          'bbbb',
          'cccc',
        ]}
      />
    );

    openMenu(wrapper);
    expect(findMenu(wrapper)).toHaveLength(1);

    act(() => {
      document.documentElement.dispatchEvent(new MouseEvent('click'));
    });
    wrapper.update();

    expect(findMenu(wrapper)).toHaveLength(0);
  });

  it('should handle menu item click', () => {
    const spy = jest.fn();

    const wrapper = mount(
      <Autocomplete
        defaultValue='0000'
        items={[
          'aaaa',
          'bbbb',
          'cccc',
          'dddd',
          'eeee',
          'ffff',
        ]}
        onSelect={spy}
      />
    );

    openMenu(wrapper);

    act(() => {
      Simulate.click(findMenuItems(wrapper).at(4).getDOMNode());
    });
    wrapper.update();

    expect(findMenu(wrapper)).toHaveLength(0);
    expect(spy).toBeCalledTimes(1);
  });

  describe('should handle field "keydown" event', () => {
    it('ArrowDown: no active item', () => {
      const spy = jest.fn();

      const wrapper = mount(
        <Autocomplete
          value='dddd'
          items={[
            'aaaa',
            'bbbb',
            'cccc',
          ]}
          onKeyDown={spy}
        />
      );

      openMenu(wrapper);
      expect(spy).toBeCalledTimes(0);

      act(() => {
        Simulate.keyDown(
          wrapper.find(TextField).find('input[data-testid="text-field:field"]').getDOMNode(),
          { key: 'ArrowDown' }
        );
      });
      wrapper.update();

      expect(spy).toBeCalledTimes(1);
      expect(isCheckedItem(findMenuItems(wrapper).at(0))).toBe(true);
      expect(isCheckedItem(findMenuItems(wrapper).at(1))).toBe(false);
      expect(isCheckedItem(findMenuItems(wrapper).at(2))).toBe(false);
    });

    it('ArrowDown: with active item', () => {
      const wrapper = mount(
        <Autocomplete
          value='dddd'
          items={[
            'aaaa',
            'bbbb',
            'cccc',
          ]}
        />
      );

      openMenu(wrapper);

      act(() => {
        Simulate.keyDown(
          wrapper.find(TextField).find('input[data-testid="text-field:field"]').getDOMNode(),
          { key: 'ArrowDown' }
        );
      });
      wrapper.update();

      // second press
      act(() => {
        Simulate.keyDown(
          wrapper.find(TextField).find('input[data-testid="text-field:field"]').getDOMNode(),
          { key: 'ArrowDown' }
        );
      });
      wrapper.update();

      expect(isCheckedItem(findMenuItems(wrapper).at(0))).toBe(false);
      expect(isCheckedItem(findMenuItems(wrapper).at(1))).toBe(true);
      expect(isCheckedItem(findMenuItems(wrapper).at(2))).toBe(false);
    });

    it('ArrowUp', () => {
      const wrapper = mount(
        <Autocomplete
          value='dddd'
          items={[
            'aaaa',
            'bbbb',
            'cccc',
          ]}
        />
      );

      openMenu(wrapper);

      act(() => {
        Simulate.keyDown(
          wrapper.find(TextField).find('input[data-testid="text-field:field"]').getDOMNode(),
          { key: 'ArrowUp' }
        );
      });
      wrapper.update();

      expect(isCheckedItem(findMenuItems(wrapper).at(0))).toBe(false);
      expect(isCheckedItem(findMenuItems(wrapper).at(1))).toBe(false);
      expect(isCheckedItem(findMenuItems(wrapper).at(2))).toBe(true);
    });

    it('Enter: no active item', () => {
      const spy = jest.fn();

      const wrapper = mount(
        <Autocomplete
          value='dddd'
          items={[
            'aaaa',
            'bbbb',
            'cccc',
          ]}
          onSelect={spy}
        />
      );

      act(() => {
        Simulate.focus(
          wrapper.find(TextField).find('input[data-testid="text-field:field"]').getDOMNode()
        );
      });
      wrapper.update();

      act(() => {
        Simulate.keyDown(
          wrapper.find(TextField).find('input[data-testid="text-field:field"]').getDOMNode(),
          { key: 'Enter' }
        );
      });
      wrapper.update();

      expect(spy).toBeCalledTimes(0);
    });

    it('Enter: with active item', () => {
      const spy = jest.fn();

      const wrapper = mount(
        <Autocomplete
          value='dddd'
          items={[
            'aaaa',
            'bbbb',
            'cccc',
          ]}
          onSelect={spy}
        />
      );

      // focus
      act(() => {
        Simulate.focus(
          wrapper.find(TextField).find('input[data-testid="text-field:field"]').getDOMNode()
        );
      });
      wrapper.update();

      // mark first as active
      act(() => {
        Simulate.keyDown(
          wrapper.find(TextField).find('input[data-testid="text-field:field"]').getDOMNode(),
          { key: 'ArrowDown' }
        );
      });
      wrapper.update();

      expect(spy).toBeCalledTimes(0);

      // press enter
      act(() => {
        Simulate.keyDown(
          wrapper.find(TextField).find('input[data-testid="text-field:field"]').getDOMNode(),
          { key: 'Enter' }
        );
      });
      wrapper.update();

      expect(spy).toBeCalledTimes(1);
    });

    it('Enter: no items', () => {
      const spy = jest.fn();

      const wrapper = mount(
        <Autocomplete
          value='dddd'
          items={undefined}
          onSelect={spy}
        />
      );

      openMenu(wrapper);

      // focus
      act(() => {
        Simulate.focus(
          wrapper.find(TextField).find('input[data-testid="text-field:field"]').getDOMNode()
        );
      });
      wrapper.update();

      // press enter
      act(() => {
        Simulate.keyDown(
          wrapper.find(TextField).find('input[data-testid="text-field:field"]').getDOMNode(),
          { key: 'Enter' }
        );
      });
      wrapper.update();

      expect(spy).toBeCalledTimes(0);
    });
  });

  it('should handle onClick prop', () => {
    const spy = jest.fn();

    const wrapper = mount(
      <Autocomplete
        items={undefined}
        onClick={spy}
      />
    );

    expect(spy).toBeCalledTimes(0);

    openMenu(wrapper);

    expect(spy).toBeCalledTimes(1);
  });
});
