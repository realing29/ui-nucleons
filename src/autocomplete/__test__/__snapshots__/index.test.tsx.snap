// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete should handle "renderItem" prop missing 1`] = `
<Autocomplete
  items={
    Array [
      "aaaa",
      "bbbb",
      "cccc",
      "dddd",
      "eeee",
    ]
  }
  value="Hello"
>
  <div
    className="root"
  >
    <ForwardRef(TextField)
      className="field"
      data-testid="autocomplete:field"
      endAdornment={
        <DownSVG
          fill="#9e9e9e"
        />
      }
      focused={true}
      multiline={false}
      onChange={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      value="Hello"
      variant="desktop"
    >
      <div
        className="root field"
        data-testid="autocomplete:field"
      >
        <div
          className="reset input-block focused variant-desktop size-l rounds-sm__all"
          data-testid="text-field:block"
          onClick={[Function]}
        >
          <div
            className="main-column"
          >
            <div
              className="input-wrapper"
            >
              <ForwardRef(BaseInput)
                className="base-input focused variant-desktop size-l"
                data-testid="text-field:field"
                multiline={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                value="Hello"
              >
                <div
                  className="reset root base-input focused variant-desktop size-l"
                >
                  <input
                    className="field reset"
                    data-testid="text-field:field"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onInput={[Function]}
                    onKeyDown={[Function]}
                    value="Hello"
                  />
                </div>
              </ForwardRef(BaseInput)>
            </div>
          </div>
          <Box
            alignItems="center"
            display="flex"
            marginLeft={4}
          >
            <div
              className="display-flex align-items-center M-l__4"
            >
              <DownSVG
                fill="#9e9e9e"
              >
                <svg
                  fill="#9e9e9e"
                  height={16}
                  viewBox="0 0 16 16"
                  width={16}
                >
                  <path
                    clipRule="evenodd"
                    d="M3.247 6.342a1 1 0 011.412-.095L8 9.171l3.341-2.924a1 1 0 011.318 1.506l-4 3.5a1 1 0 01-1.317 0l-4-3.5a1 1 0 01-.095-1.411z"
                    fillRule="evenodd"
                  />
                </svg>
              </DownSVG>
            </div>
          </Box>
        </div>
      </div>
    </ForwardRef(TextField)>
    <ForwardRef(Dropdown)
      className="menu"
      data-testid="autocomplete:menu"
      role="menu"
      style={
        Object {
          "top": "56px",
        }
      }
    >
      <div
        className="root menu shadow-z4 rounds-md__all"
        data-testid="autocomplete:menu"
        role="menu"
        style={
          Object {
            "top": "56px",
          }
        }
      >
        <CustomScrollbar
          className="inner"
          overflow={
            Object {
              "x": "h",
              "y": "s",
            }
          }
        >
          <OverlayScrollbarsComponent
            className="custom-scrollbar inner"
            options={
              Object {
                "overflowBehavior": Object {
                  "x": "h",
                  "y": "s",
                },
              }
            }
          >
            <div
              className="os-host"
            >
              <div
                className="os-resize-observer-host"
              />
              <div
                className="os-padding"
              >
                <div
                  className="os-viewport"
                >
                  <div
                    className="os-content"
                  >
                    <DropdownItem
                      checked={false}
                      key="0"
                      noHover={true}
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      role="menuitem"
                    >
                      <div
                        className="root size-m no-hover"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        role="menuitem"
                      >
                        aaaa
                      </div>
                    </DropdownItem>
                    <DropdownItem
                      checked={false}
                      key="1"
                      noHover={true}
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      role="menuitem"
                    >
                      <div
                        className="root size-m no-hover"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        role="menuitem"
                      >
                        bbbb
                      </div>
                    </DropdownItem>
                    <DropdownItem
                      checked={false}
                      key="2"
                      noHover={true}
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      role="menuitem"
                    >
                      <div
                        className="root size-m no-hover"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        role="menuitem"
                      >
                        cccc
                      </div>
                    </DropdownItem>
                    <DropdownItem
                      checked={false}
                      key="3"
                      noHover={true}
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      role="menuitem"
                    >
                      <div
                        className="root size-m no-hover"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        role="menuitem"
                      >
                        dddd
                      </div>
                    </DropdownItem>
                    <DropdownItem
                      checked={false}
                      key="4"
                      noHover={true}
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      role="menuitem"
                    >
                      <div
                        className="root size-m no-hover"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        role="menuitem"
                      >
                        eeee
                      </div>
                    </DropdownItem>
                  </div>
                </div>
              </div>
              <div
                className="os-scrollbar os-scrollbar-horizontal "
              >
                <div
                  className="os-scrollbar-track"
                >
                  <div
                    className="os-scrollbar-handle"
                  />
                </div>
              </div>
              <div
                className="os-scrollbar os-scrollbar-vertical"
              >
                <div
                  className="os-scrollbar-track"
                >
                  <div
                    className="os-scrollbar-handle"
                  />
                </div>
              </div>
              <div
                className="os-scrollbar-corner"
              />
            </div>
          </OverlayScrollbarsComponent>
        </CustomScrollbar>
      </div>
    </ForwardRef(Dropdown)>
  </div>
</Autocomplete>
`;

exports[`Autocomplete should handle defaultValue 1`] = `
<Autocomplete
  defaultValue="dddd"
  items={
    Array [
      "aaaa",
      "bbbb",
      "cccc",
    ]
  }
>
  <div
    className="root"
  >
    <ForwardRef(TextField)
      className="field"
      data-testid="autocomplete:field"
      endAdornment={
        <DownSVG
          fill="#9e9e9e"
        />
      }
      focused={false}
      multiline={false}
      onChange={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      variant="desktop"
    >
      <div
        className="root field"
        data-testid="autocomplete:field"
      >
        <div
          className="reset input-block variant-desktop size-l rounds-sm__all"
          data-testid="text-field:block"
          onClick={[Function]}
        >
          <div
            className="main-column"
          >
            <div
              className="input-wrapper"
            >
              <ForwardRef(BaseInput)
                className="base-input variant-desktop size-l"
                data-testid="text-field:field"
                multiline={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
              >
                <div
                  className="reset root base-input variant-desktop size-l"
                >
                  <input
                    className="field reset"
                    data-testid="text-field:field"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onInput={[Function]}
                    onKeyDown={[Function]}
                  />
                </div>
              </ForwardRef(BaseInput)>
            </div>
          </div>
          <Box
            alignItems="center"
            display="flex"
            marginLeft={4}
          >
            <div
              className="display-flex align-items-center M-l__4"
            >
              <DownSVG
                fill="#9e9e9e"
              >
                <svg
                  fill="#9e9e9e"
                  height={16}
                  viewBox="0 0 16 16"
                  width={16}
                >
                  <path
                    clipRule="evenodd"
                    d="M3.247 6.342a1 1 0 011.412-.095L8 9.171l3.341-2.924a1 1 0 011.318 1.506l-4 3.5a1 1 0 01-1.317 0l-4-3.5a1 1 0 01-.095-1.411z"
                    fillRule="evenodd"
                  />
                </svg>
              </DownSVG>
            </div>
          </Box>
        </div>
      </div>
    </ForwardRef(TextField)>
  </div>
</Autocomplete>
`;

exports[`Autocomplete should renders correctly 1`] = `
<Autocomplete
  items={
    Array [
      "U.S.",
      "France",
      "China",
      "Cambodia",
      "Chile",
      "Canada",
      "Poland",
      "Russia",
      "Belarus",
      "Ukraine",
    ]
  }
  renderItem={[Function]}
  value="Hello"
>
  <div
    className="root"
  >
    <ForwardRef(TextField)
      className="field"
      data-testid="autocomplete:field"
      endAdornment={
        <DownSVG
          fill="#9e9e9e"
        />
      }
      focused={false}
      multiline={false}
      onChange={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      value="Hello"
      variant="desktop"
    >
      <div
        className="root field"
        data-testid="autocomplete:field"
      >
        <div
          className="reset input-block variant-desktop size-l rounds-sm__all"
          data-testid="text-field:block"
          onClick={[Function]}
        >
          <div
            className="main-column"
          >
            <div
              className="input-wrapper"
            >
              <ForwardRef(BaseInput)
                className="base-input variant-desktop size-l"
                data-testid="text-field:field"
                multiline={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                value="Hello"
              >
                <div
                  className="reset root base-input variant-desktop size-l"
                >
                  <input
                    className="field reset"
                    data-testid="text-field:field"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onInput={[Function]}
                    onKeyDown={[Function]}
                    value="Hello"
                  />
                </div>
              </ForwardRef(BaseInput)>
            </div>
          </div>
          <Box
            alignItems="center"
            display="flex"
            marginLeft={4}
          >
            <div
              className="display-flex align-items-center M-l__4"
            >
              <DownSVG
                fill="#9e9e9e"
              >
                <svg
                  fill="#9e9e9e"
                  height={16}
                  viewBox="0 0 16 16"
                  width={16}
                >
                  <path
                    clipRule="evenodd"
                    d="M3.247 6.342a1 1 0 011.412-.095L8 9.171l3.341-2.924a1 1 0 011.318 1.506l-4 3.5a1 1 0 01-1.317 0l-4-3.5a1 1 0 01-.095-1.411z"
                    fillRule="evenodd"
                  />
                </svg>
              </DownSVG>
            </div>
          </Box>
        </div>
      </div>
    </ForwardRef(TextField)>
  </div>
</Autocomplete>
`;

exports[`Autocomplete should renders correctly 2`] = `
<Autocomplete
  items={
    Array [
      "U.S.",
      "France",
      "China",
      "Cambodia",
      "Chile",
      "Canada",
      "Poland",
      "Russia",
      "Belarus",
      "Ukraine",
    ]
  }
  preset="filled-only-list"
  renderItem={[Function]}
  value="Hello"
>
  <div
    className="root"
  >
    <ForwardRef(TextField)
      className="field"
      data-testid="autocomplete:field"
      focused={false}
      multiline={false}
      onChange={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      value="Hello"
      variant="desktop"
    >
      <div
        className="root field"
        data-testid="autocomplete:field"
      >
        <div
          className="reset input-block variant-desktop size-l rounds-sm__all"
          data-testid="text-field:block"
          onClick={[Function]}
        >
          <div
            className="main-column"
          >
            <div
              className="input-wrapper"
            >
              <ForwardRef(BaseInput)
                className="base-input variant-desktop size-l"
                data-testid="text-field:field"
                multiline={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                value="Hello"
              >
                <div
                  className="reset root base-input variant-desktop size-l"
                >
                  <input
                    className="field reset"
                    data-testid="text-field:field"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onInput={[Function]}
                    onKeyDown={[Function]}
                    value="Hello"
                  />
                </div>
              </ForwardRef(BaseInput)>
            </div>
          </div>
        </div>
      </div>
    </ForwardRef(TextField)>
  </div>
</Autocomplete>
`;

exports[`Autocomplete should renders correctly 3`] = `
<Autocomplete
  items={
    Array [
      "U.S.",
      "France",
      "China",
      "Cambodia",
      "Chile",
      "Canada",
      "Poland",
      "Russia",
      "Belarus",
      "Ukraine",
    ]
  }
  preset="filled-only-list"
  renderItem={[Function]}
  value="Hello"
>
  <div
    className="root"
  >
    <ForwardRef(TextField)
      className="field"
      data-testid="autocomplete:field"
      focused={true}
      multiline={false}
      onChange={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      value="Hello"
      variant="desktop"
    >
      <div
        className="root field"
        data-testid="autocomplete:field"
      >
        <div
          className="reset input-block focused variant-desktop size-l rounds-sm__all"
          data-testid="text-field:block"
          onClick={[Function]}
        >
          <div
            className="main-column"
          >
            <div
              className="input-wrapper"
            >
              <ForwardRef(BaseInput)
                className="base-input focused variant-desktop size-l"
                data-testid="text-field:field"
                multiline={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                value="Hello"
              >
                <div
                  className="reset root base-input focused variant-desktop size-l"
                >
                  <input
                    className="field reset"
                    data-testid="text-field:field"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onInput={[Function]}
                    onKeyDown={[Function]}
                    value="Hello"
                  />
                </div>
              </ForwardRef(BaseInput)>
            </div>
          </div>
        </div>
      </div>
    </ForwardRef(TextField)>
    <ForwardRef(Dropdown)
      className="menu"
      data-testid="autocomplete:menu"
      role="menu"
      style={
        Object {
          "top": "56px",
        }
      }
    >
      <div
        className="root menu shadow-z4 rounds-md__all"
        data-testid="autocomplete:menu"
        role="menu"
        style={
          Object {
            "top": "56px",
          }
        }
      >
        <CustomScrollbar
          className="inner"
          overflow={
            Object {
              "x": "h",
              "y": "s",
            }
          }
        >
          <OverlayScrollbarsComponent
            className="custom-scrollbar inner"
            options={
              Object {
                "overflowBehavior": Object {
                  "x": "h",
                  "y": "s",
                },
              }
            }
          >
            <div
              className="os-host"
            >
              <div
                className="os-resize-observer-host"
              />
              <div
                className="os-padding"
              >
                <div
                  className="os-viewport"
                >
                  <div
                    className="os-content"
                  >
                    <DropdownItem
                      checked={false}
                      key="0"
                      noHover={true}
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      role="menuitem"
                    >
                      <div
                        className="root size-m no-hover"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        role="menuitem"
                      >
                        U.S.
                      </div>
                    </DropdownItem>
                    <DropdownItem
                      checked={false}
                      key="1"
                      noHover={true}
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      role="menuitem"
                    >
                      <div
                        className="root size-m no-hover"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        role="menuitem"
                      >
                        FRANCE
                      </div>
                    </DropdownItem>
                    <DropdownItem
                      checked={false}
                      key="2"
                      noHover={true}
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      role="menuitem"
                    >
                      <div
                        className="root size-m no-hover"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        role="menuitem"
                      >
                        CHINA
                      </div>
                    </DropdownItem>
                    <DropdownItem
                      checked={false}
                      key="3"
                      noHover={true}
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      role="menuitem"
                    >
                      <div
                        className="root size-m no-hover"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        role="menuitem"
                      >
                        CAMBODIA
                      </div>
                    </DropdownItem>
                    <DropdownItem
                      checked={false}
                      key="4"
                      noHover={true}
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      role="menuitem"
                    >
                      <div
                        className="root size-m no-hover"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        role="menuitem"
                      >
                        CHILE
                      </div>
                    </DropdownItem>
                    <DropdownItem
                      checked={false}
                      key="5"
                      noHover={true}
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      role="menuitem"
                    >
                      <div
                        className="root size-m no-hover"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        role="menuitem"
                      >
                        CANADA
                      </div>
                    </DropdownItem>
                    <DropdownItem
                      checked={false}
                      key="6"
                      noHover={true}
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      role="menuitem"
                    >
                      <div
                        className="root size-m no-hover"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        role="menuitem"
                      >
                        POLAND
                      </div>
                    </DropdownItem>
                    <DropdownItem
                      checked={false}
                      key="7"
                      noHover={true}
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      role="menuitem"
                    >
                      <div
                        className="root size-m no-hover"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        role="menuitem"
                      >
                        RUSSIA
                      </div>
                    </DropdownItem>
                    <DropdownItem
                      checked={false}
                      key="8"
                      noHover={true}
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      role="menuitem"
                    >
                      <div
                        className="root size-m no-hover"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        role="menuitem"
                      >
                        BELARUS
                      </div>
                    </DropdownItem>
                    <DropdownItem
                      checked={false}
                      key="9"
                      noHover={true}
                      onClick={[Function]}
                      onMouseEnter={[Function]}
                      role="menuitem"
                    >
                      <div
                        className="root size-m no-hover"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        role="menuitem"
                      >
                        UKRAINE
                      </div>
                    </DropdownItem>
                  </div>
                </div>
              </div>
              <div
                className="os-scrollbar os-scrollbar-horizontal "
              >
                <div
                  className="os-scrollbar-track"
                >
                  <div
                    className="os-scrollbar-handle"
                  />
                </div>
              </div>
              <div
                className="os-scrollbar os-scrollbar-vertical"
              >
                <div
                  className="os-scrollbar-track"
                >
                  <div
                    className="os-scrollbar-handle"
                  />
                </div>
              </div>
              <div
                className="os-scrollbar-corner"
              />
            </div>
          </OverlayScrollbarsComponent>
        </CustomScrollbar>
      </div>
    </ForwardRef(Dropdown)>
  </div>
</Autocomplete>
`;

exports[`Autocomplete should renders correctly 4`] = `
<Autocomplete
  items={
    Array [
      "U.S.",
      "France",
      "China",
      "Cambodia",
      "Chile",
      "Canada",
      "Poland",
      "Russia",
      "Belarus",
      "Ukraine",
    ]
  }
  loading={true}
  preset="filled-only-list"
  renderItem={[Function]}
  value="Hello"
>
  <div
    className="root"
  >
    <ForwardRef(TextField)
      className="field"
      data-testid="autocomplete:field"
      focused={true}
      multiline={false}
      onChange={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      value="Hello"
      variant="desktop"
    >
      <div
        className="root field"
        data-testid="autocomplete:field"
      >
        <div
          className="reset input-block focused variant-desktop size-l rounds-sm__all"
          data-testid="text-field:block"
          onClick={[Function]}
        >
          <div
            className="main-column"
          >
            <div
              className="input-wrapper"
            >
              <ForwardRef(BaseInput)
                className="base-input focused variant-desktop size-l"
                data-testid="text-field:field"
                multiline={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                value="Hello"
              >
                <div
                  className="reset root base-input focused variant-desktop size-l"
                >
                  <input
                    className="field reset"
                    data-testid="text-field:field"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onInput={[Function]}
                    onKeyDown={[Function]}
                    value="Hello"
                  />
                </div>
              </ForwardRef(BaseInput)>
            </div>
          </div>
        </div>
      </div>
    </ForwardRef(TextField)>
    <ForwardRef(Dropdown)
      className="menu"
      data-testid="autocomplete:menu"
      role="menu"
      style={
        Object {
          "top": "56px",
        }
      }
    >
      <div
        className="root menu shadow-z4 rounds-md__all"
        data-testid="autocomplete:menu"
        role="menu"
        style={
          Object {
            "top": "56px",
          }
        }
      >
        <CustomScrollbar
          className="inner"
          overflow={
            Object {
              "x": "h",
              "y": "s",
            }
          }
        >
          <OverlayScrollbarsComponent
            className="custom-scrollbar inner"
            options={
              Object {
                "overflowBehavior": Object {
                  "x": "h",
                  "y": "s",
                },
              }
            }
          >
            <div
              className="os-host"
            >
              <div
                className="os-resize-observer-host"
              />
              <div
                className="os-padding"
              >
                <div
                  className="os-viewport"
                >
                  <div
                    className="os-content"
                  >
                    <DropdownLoading
                      data-testid="autocomplete:loading-area"
                    >
                      <div
                        className="root"
                        data-testid="autocomplete:loading-area"
                      >
                        <Spinner
                          size="small"
                        >
                          <div>
                            <svg
                              className="spinner size-small"
                              height={20}
                              stroke="#1f84db"
                              viewBox="0 0 20 20"
                              width={20}
                            >
                              <circle
                                className="circle"
                                cx={10}
                                cy={10}
                                r={9}
                              />
                            </svg>
                          </div>
                        </Spinner>
                      </div>
                    </DropdownLoading>
                  </div>
                </div>
              </div>
              <div
                className="os-scrollbar os-scrollbar-horizontal "
              >
                <div
                  className="os-scrollbar-track"
                >
                  <div
                    className="os-scrollbar-handle"
                  />
                </div>
              </div>
              <div
                className="os-scrollbar os-scrollbar-vertical"
              >
                <div
                  className="os-scrollbar-track"
                >
                  <div
                    className="os-scrollbar-handle"
                  />
                </div>
              </div>
              <div
                className="os-scrollbar-corner"
              />
            </div>
          </OverlayScrollbarsComponent>
        </CustomScrollbar>
      </div>
    </ForwardRef(Dropdown)>
  </div>
</Autocomplete>
`;
